### PR TITLE
Add architecture and sequence diagrams

### PR DESCRIPTION
**Pull request by cam-schultz**
## Why this should be merged
Adds architecture and sequence diagrams pertaining to online message relaying. Does not cover the catch up mechanism or manual Warp messages for simplicity.

## How is this documented
Adds mermaid diagrams

### CODE REVIEW

**Review by bernard-avalabs — COMMENTED** [review]
The sequence diagram is very helpful. I left a comment about one of the transitions.

File: docs/sequence.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    participant AppRequestNetwork
+    participant DestinationClient
+    participant CheckpointManager
+    participant RelayerDatabase
+
+    Subscriber->>Listener : (async) New block
+    activate Listener
+    Listener->>Listener : Collect Warp Logs
+    Listener->>Listener : Create message handlers
+    Listener-->>ApplicationRelayer : (async) Message handlers for block
```
[the remark this answers is not in the document]
Comment (cam-schultz) [reply docs/sequence.md:18]: > Is the Listener asynchronously sending the Message handlers to the ApplicationRelayer in this step?

Yes, that's exactly right. I updated the wording to hopefully make this a bit clearer.

The dotted lines are meant to represent asynchronous calls, but I'm happy to amend if there's a more standard convention.

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Architecture
+Partial class diagram illustrating the important classes and functions for processing incoming blocks and relaying the contained Warp messages.
+```mermaid
+classDiagram
+    Listener o-- "per RelayerID" ApplicationRelayer
+    Listener *-- "1" Subscriber
```
Comment (bernard-avalabs) [inline docs/architecture.md:6]: What does the "1" represent here in the composition link?

**Review by bernard-avalabs — COMMENTED** [review]
Generally makes sense to me. I left a comment about one of the composition links in the UML diagram. I don't draw/read UML diagrams that often, so perhaps my question would be obvious to someone else who does.

File: docs/architecture.md (hunk 2 above)
Comment (cam-schultz) [reply docs/architecture.md:6]: For clarity, the same notation applies to compositions as well as aggregations. In the above example, `Listeners` aggregate `ApplicationRelayers`

**Review by geoff-vball — APPROVED** [review]

**Review by iansuvak — APPROVED** [review]
LGTM and thanks for making this. This is very helpful to have while studying the codebase

**Pull request merged**